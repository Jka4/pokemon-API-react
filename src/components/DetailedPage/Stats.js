import React from "react";

const Stats = (props) => {
  const { stats, weight } = props;

  return (
    <div className="stats">
      <ul className="statsUL">
        <div className="skills_title">Stats</div>

        {stats &&
          stats.map((i, key) => (
            <li className="statsLI" key={key}>
              {i.stat.name} {i.base_stat}
            </li>
          ))}
        <li className="statsLI">weight {weight}kg</li>
      </ul>
    </div>
  );
};

export default Stats;
