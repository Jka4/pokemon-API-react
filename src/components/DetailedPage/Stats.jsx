import React from "react";
import Paper from "@material-ui/core/Paper";

// interface statsTypes {
//   base_stat?: number,
//   effort?: number,
//   stat: {
//     name?: string,
//     url?: string,
//   }
// }

const Stats = ({ stats, weight }) => {
  return (
    <div className="stats">
      <Paper elevation={3} className="statsUL">
        <div className="skills_title">Stats</div>
        {stats.map((element, key) => (
          <li className="statsLI" key={key}>
            {element.stat.name} {element.base_stat}
          </li>
        ))}
        <li className="statsLI">weight {weight}kg</li>
      </Paper>
    </div>
  );
};

export default Stats;
