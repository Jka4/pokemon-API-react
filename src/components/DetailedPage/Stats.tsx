import React from "react";
import Paper from "@material-ui/core/Paper";

interface StatsElements {
  el: {
    base_stat: number;
    stat: {
      name: string;
    };
  };
  [key: string]: any;
}

interface Props {
  stats: StatsElements[];
  weight: number;
}

const Stats: React.FC<Props> = ({ stats, weight }: Props) => {
  return (
    <div className="stats">
      <Paper elevation={3} className="statsUL">
        <div className="skills_title">Stats</div>
        {stats &&
          stats.map((el: StatsElements) => (
            <li className="statsLI" key={el?.stat?.name}>
              {el?.stat?.name} {el?.base_stat}
            </li>
          ))}
        <li className="statsLI">weight {weight}kg</li>
      </Paper>
    </div>
  );
};

export default Stats;
