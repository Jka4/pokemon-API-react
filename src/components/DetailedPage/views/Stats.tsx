import React from 'react';
import styled from 'styled-components';
import { Skeleton } from '@material-ui/lab';

interface Props {
  stats: StatsElements[];
  weight: number;
}

interface StatsElements {
  base_stat: number;
  stat: {
    name: string;
  };
}

const Stats: React.FC<Props> = ({ stats, weight }: Props) => {
  const fallbackSkeletons = () => {
    const fakeArr = Array.from(Array(7).keys());

    return (
      <>
        {fakeArr.map((el) => (
          <LI key={el}>
            <Skeleton animation="wave" width={135} height={30} />{' '}
          </LI>
        ))}
      </>
    );
  };

  const abilitiesRow = () => {
    return (
      <>
        {stats.map((el: StatsElements) => (
          <LI key={el.stat.name}>
            {el.stat.name}: {el.base_stat}
          </LI>
        ))}
        <LI>weight: {weight} kg</LI>
      </>
    );
  };

  const showSkeletons = stats.length === 0;

  return (
    <StatsStyled>
      <UL>
        <Title>Stats</Title>
        {showSkeletons ? fallbackSkeletons() : abilitiesRow()}
      </UL>
    </StatsStyled>
  );
};

const StatsStyled = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  margin-right: 3rem;
  width: 200px;
  height: 380px;
`;

const Title = styled.div`
  align-self: center;
  justify-self: center;
  font-size: 2rem;
`;

const UL = styled.ul`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-start;
  border: 1px solid black;
  border-radius: 0.7rem;
  background-color: #53aeff;
  padding: 25px;
  width: 100%;
  height: 100%;
  margin-top: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    padding: 10px;
  }
`;

const LI = styled.li`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-start;
  padding: 6px;
  width: 100%;
  font-size: 18px;
  text-transform: uppercase;

  @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    font-size: 14px;
    padding-bottom: 5px;
  }
`;

export default Stats;
