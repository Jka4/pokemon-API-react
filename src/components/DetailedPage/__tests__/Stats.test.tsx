import Stats from '../views/Stats';

import { mount } from 'enzyme';

describe('<Stats/> ', () => {
  it('renders correctly', () => {
    const fakeData = [
      {
        base_stat: 70,
        stat: {
          name: 'hp',
          url: 'https://pokeapi.co/api/v2/stat/1/',
        },
      },
      {
        base_stat: 40,
        stat: {
          name: 'speed',
          url: 'https://pokeapi.co/api/v2/stat/6/',
        },
      },
    ];

    const container = mount(<Stats stats={fakeData} weight={123} />);

    expect(container).toMatchSnapshot();
  });

  it('without data', () => {
    const container = mount(<Stats stats={[]} weight={123} />);
    expect(container).toMatchSnapshot();
  });
});
