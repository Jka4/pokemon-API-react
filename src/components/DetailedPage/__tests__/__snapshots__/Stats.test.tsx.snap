// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stats/>  renders correctly 1`] = `
<Stats
  stats={
    Array [
      Object {
        "base_stat": 70,
        "stat": Object {
          "name": "hp",
          "url": "https://pokeapi.co/api/v2/stat/1/",
        },
      },
      Object {
        "base_stat": 40,
        "stat": Object {
          "name": "speed",
          "url": "https://pokeapi.co/api/v2/stat/6/",
        },
      },
    ]
  }
  weight={123}
>
  <styled.div>
    <div
      className="sc-bdfBwQ cPDFDP"
    >
      <styled.ul>
        <ul
          className="sc-dlfnbm fthlrP"
        >
          <styled.div>
            <div
              className="sc-gsTCUz GIzOt"
            >
              Stats
            </div>
          </styled.div>
          <styled.li
            key="hp"
          >
            <li
              className="sc-hKgILt kPLVMO"
            >
              hp
              : 
              70
            </li>
          </styled.li>
          <styled.li
            key="speed"
          >
            <li
              className="sc-hKgILt kPLVMO"
            >
              speed
              : 
              40
            </li>
          </styled.li>
          <styled.li>
            <li
              className="sc-hKgILt kPLVMO"
            >
              weight: 
              123
               kg
            </li>
          </styled.li>
        </ul>
      </styled.ul>
    </div>
  </styled.div>
</Stats>
`;

exports[`<Stats/>  without data 1`] = `
<Stats
  stats={Array []}
  weight={123}
>
  <styled.div>
    <div
      className="sc-bdfBwQ cPDFDP"
    >
      <styled.ul>
        <ul
          className="sc-dlfnbm fthlrP"
        >
          <styled.div>
            <div
              className="sc-gsTCUz GIzOt"
            >
              Stats
            </div>
          </styled.div>
          <styled.li
            key="0"
          >
            <li
              className="sc-hKgILt kPLVMO"
            >
              <WithStyles(ForwardRef(Skeleton))
                animation="wave"
                height={30}
                width={135}
              >
                <ForwardRef(Skeleton)
                  animation="wave"
                  classes={
                    Object {
                      "circle": "MuiSkeleton-circle",
                      "fitContent": "MuiSkeleton-fitContent",
                      "heightAuto": "MuiSkeleton-heightAuto",
                      "pulse": "MuiSkeleton-pulse",
                      "rect": "MuiSkeleton-rect",
                      "root": "MuiSkeleton-root",
                      "text": "MuiSkeleton-text",
                      "wave": "MuiSkeleton-wave",
                      "withChildren": "MuiSkeleton-withChildren",
                    }
                  }
                  height={30}
                  width={135}
                >
                  <span
                    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                    style={
                      Object {
                        "height": 30,
                        "width": 135,
                      }
                    }
                  />
                </ForwardRef(Skeleton)>
              </WithStyles(ForwardRef(Skeleton))>
               
            </li>
          </styled.li>
          <styled.li
            key="1"
          >
            <li
              className="sc-hKgILt kPLVMO"
            >
              <WithStyles(ForwardRef(Skeleton))
                animation="wave"
                height={30}
                width={135}
              >
                <ForwardRef(Skeleton)
                  animation="wave"
                  classes={
                    Object {
                      "circle": "MuiSkeleton-circle",
                      "fitContent": "MuiSkeleton-fitContent",
                      "heightAuto": "MuiSkeleton-heightAuto",
                      "pulse": "MuiSkeleton-pulse",
                      "rect": "MuiSkeleton-rect",
                      "root": "MuiSkeleton-root",
                      "text": "MuiSkeleton-text",
                      "wave": "MuiSkeleton-wave",
                      "withChildren": "MuiSkeleton-withChildren",
                    }
                  }
                  height={30}
                  width={135}
                >
                  <span
                    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                    style={
                      Object {
                        "height": 30,
                        "width": 135,
                      }
                    }
                  />
                </ForwardRef(Skeleton)>
              </WithStyles(ForwardRef(Skeleton))>
               
            </li>
          </styled.li>
          <styled.li
            key="2"
          >
            <li
              className="sc-hKgILt kPLVMO"
            >
              <WithStyles(ForwardRef(Skeleton))
                animation="wave"
                height={30}
                width={135}
              >
                <ForwardRef(Skeleton)
                  animation="wave"
                  classes={
                    Object {
                      "circle": "MuiSkeleton-circle",
                      "fitContent": "MuiSkeleton-fitContent",
                      "heightAuto": "MuiSkeleton-heightAuto",
                      "pulse": "MuiSkeleton-pulse",
                      "rect": "MuiSkeleton-rect",
                      "root": "MuiSkeleton-root",
                      "text": "MuiSkeleton-text",
                      "wave": "MuiSkeleton-wave",
                      "withChildren": "MuiSkeleton-withChildren",
                    }
                  }
                  height={30}
                  width={135}
                >
                  <span
                    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                    style={
                      Object {
                        "height": 30,
                        "width": 135,
                      }
                    }
                  />
                </ForwardRef(Skeleton)>
              </WithStyles(ForwardRef(Skeleton))>
               
            </li>
          </styled.li>
          <styled.li
            key="3"
          >
            <li
              className="sc-hKgILt kPLVMO"
            >
              <WithStyles(ForwardRef(Skeleton))
                animation="wave"
                height={30}
                width={135}
              >
                <ForwardRef(Skeleton)
                  animation="wave"
                  classes={
                    Object {
                      "circle": "MuiSkeleton-circle",
                      "fitContent": "MuiSkeleton-fitContent",
                      "heightAuto": "MuiSkeleton-heightAuto",
                      "pulse": "MuiSkeleton-pulse",
                      "rect": "MuiSkeleton-rect",
                      "root": "MuiSkeleton-root",
                      "text": "MuiSkeleton-text",
                      "wave": "MuiSkeleton-wave",
                      "withChildren": "MuiSkeleton-withChildren",
                    }
                  }
                  height={30}
                  width={135}
                >
                  <span
                    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                    style={
                      Object {
                        "height": 30,
                        "width": 135,
                      }
                    }
                  />
                </ForwardRef(Skeleton)>
              </WithStyles(ForwardRef(Skeleton))>
               
            </li>
          </styled.li>
          <styled.li
            key="4"
          >
            <li
              className="sc-hKgILt kPLVMO"
            >
              <WithStyles(ForwardRef(Skeleton))
                animation="wave"
                height={30}
                width={135}
              >
                <ForwardRef(Skeleton)
                  animation="wave"
                  classes={
                    Object {
                      "circle": "MuiSkeleton-circle",
                      "fitContent": "MuiSkeleton-fitContent",
                      "heightAuto": "MuiSkeleton-heightAuto",
                      "pulse": "MuiSkeleton-pulse",
                      "rect": "MuiSkeleton-rect",
                      "root": "MuiSkeleton-root",
                      "text": "MuiSkeleton-text",
                      "wave": "MuiSkeleton-wave",
                      "withChildren": "MuiSkeleton-withChildren",
                    }
                  }
                  height={30}
                  width={135}
                >
                  <span
                    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                    style={
                      Object {
                        "height": 30,
                        "width": 135,
                      }
                    }
                  />
                </ForwardRef(Skeleton)>
              </WithStyles(ForwardRef(Skeleton))>
               
            </li>
          </styled.li>
          <styled.li
            key="5"
          >
            <li
              className="sc-hKgILt kPLVMO"
            >
              <WithStyles(ForwardRef(Skeleton))
                animation="wave"
                height={30}
                width={135}
              >
                <ForwardRef(Skeleton)
                  animation="wave"
                  classes={
                    Object {
                      "circle": "MuiSkeleton-circle",
                      "fitContent": "MuiSkeleton-fitContent",
                      "heightAuto": "MuiSkeleton-heightAuto",
                      "pulse": "MuiSkeleton-pulse",
                      "rect": "MuiSkeleton-rect",
                      "root": "MuiSkeleton-root",
                      "text": "MuiSkeleton-text",
                      "wave": "MuiSkeleton-wave",
                      "withChildren": "MuiSkeleton-withChildren",
                    }
                  }
                  height={30}
                  width={135}
                >
                  <span
                    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                    style={
                      Object {
                        "height": 30,
                        "width": 135,
                      }
                    }
                  />
                </ForwardRef(Skeleton)>
              </WithStyles(ForwardRef(Skeleton))>
               
            </li>
          </styled.li>
          <styled.li
            key="6"
          >
            <li
              className="sc-hKgILt kPLVMO"
            >
              <WithStyles(ForwardRef(Skeleton))
                animation="wave"
                height={30}
                width={135}
              >
                <ForwardRef(Skeleton)
                  animation="wave"
                  classes={
                    Object {
                      "circle": "MuiSkeleton-circle",
                      "fitContent": "MuiSkeleton-fitContent",
                      "heightAuto": "MuiSkeleton-heightAuto",
                      "pulse": "MuiSkeleton-pulse",
                      "rect": "MuiSkeleton-rect",
                      "root": "MuiSkeleton-root",
                      "text": "MuiSkeleton-text",
                      "wave": "MuiSkeleton-wave",
                      "withChildren": "MuiSkeleton-withChildren",
                    }
                  }
                  height={30}
                  width={135}
                >
                  <span
                    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                    style={
                      Object {
                        "height": 30,
                        "width": 135,
                      }
                    }
                  />
                </ForwardRef(Skeleton)>
              </WithStyles(ForwardRef(Skeleton))>
               
            </li>
          </styled.li>
        </ul>
      </styled.ul>
    </div>
  </styled.div>
</Stats>
`;
